server/protocols/jmap/src/main/java/org/apache/james/jmap/JMAPServlet.java

```diff
@@ -18,92 +18,68 @@
-import org.apache.james.jmap.json.MultipleObjectMapperBuilder;
-import org.apache.james.jmap.model.AccessTokenRequest;
-import org.apache.james.jmap.model.AccessTokenResponse;
-import org.apache.james.jmap.model.ContinuationTokenRequest;
-import org.apache.james.jmap.model.ContinuationTokenResponse;
-import org.apache.james.user.api.UsersRepository;
-import org.apache.james.user.api.UsersRepositoryException;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import org.apache.james.jmap.methods.MethodDispatcher;
+import org.apache.james.jmap.model.Protocol;
 
-import com.fasterxml.jackson.core.JsonParseException;
-import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.google.common.annotations.VisibleForTesting;
-import com.google.common.io.ByteStreams;
 import com.google.common.io.CharStreams;
 
 public class JMAPServlet extends HttpServlet {
 
-    public static class RawRequest {
-        public Object element;
-
-        public RawRequest(Object element) {
-            super();
-            this.element = element;
-        }
-
-        public RawRequest(String element) {
-            super();
-            this.element = element;
-        }
+    private static final long serialVersionUID = 1L;
 
-        public RawRequest(GetAccountsRequest element) {
-            super();
-            this.element = element;
-        }
-        
-        
-    }
-    
-    public static class Request {
-        public String action = "getAccounts";
-        public GetAccountsRequest arguments;
-        public String clientId;
-    }
-    
-    public static class GetAccountsRequest {
-        public String state;
-    }
-    
     public static final String JSON_CONTENT_TYPE = "application/json";
     public static final String JSON_CONTENT_TYPE_UTF8 = "application/json; charset=UTF-8";
 
-    private static final Logger LOG = LoggerFactory.getLogger(JMAPServlet.class);
-
     private ObjectMapper objectMapper = new ObjectMapper();
-    
-    private UsersRepository usersRepository;
+    private MethodDispatcher methodDispatcher;
 
     @Override
-    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
-        String username = "roger@barcamp";
-        String action = "getAccounts";
-        String input = CharStreams.toString(new InputStreamReader(req.getInputStream()));
-        
-        try (OutputStreamWriter output = new OutputStreamWriter(resp.getOutputStream())) {
-            output.append("[\"error\", {type: \"invalidArgument\"}, \"client-id\"]");
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException {
+        try {
+            String input = CharStreams.toString(new InputStreamReader(req.getInputStream()));
+            List<Protocol> requests = deserialize(input);
```

Comment:
How do you prevent a request to be used as a non-processed response with both using Protocol type ? I'd rather use Request and Response in types.